### PR TITLE
mds: do not send split_realms for CEPH_SNAP_OP_UPDATE msg

### PR DESCRIPTION
**Pull request by lxbsz**
The clients won't care about the split_realms and the kclient will treat this as a corrupted snaptrace.

Fixes commit 93e7267757508520dfc22cff1ab20558bd4a44d4 ("mds: send snap
	related messages centrally during mds recovery")
Fixes: https://tracker.ceph.com/issues/61217
Signed-off-by: Xiubo Li <xiubli@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by joscollin — APPROVED** [review]

File: src/mds/MDCache.cc [hunk 1]
```diff
@@ -6037,8 +6037,6 @@ void MDCache::finish_snaprealm_reconnect(client_t client, SnapRealm *realm, snap
 	     << realm->get_newest_seq() << " on " << *realm << dendl;
     auto snap = make_message<MClientSnap>(CEPH_SNAP_OP_UPDATE);
     snap->bl = mds->server->get_snap_trace(client, realm);
-    for (const auto& child : realm->open_children)
-      snap->split_realms.push_back(child->inode->ino());
```
[the remark this answers is not in the document]
Comment (lxbsz) [reply src/mds/MDCache.cc:6041]: yeah, Zheng just copied the code from the old split op case in commit: https://github.com/ceph/ceph/commit/93e7267757508520dfc22cff1ab20558bd4a44d4 by mistake.

[the remark this answers is not in the document]
Comment (dparmar18) [reply src/mds/MDCache.cc:6041]: alright

**Comment by vshankar** [comment]
@lxbsz Please help me recall - the corrupted snaptrace issue was also supposed to be fixed in the clients too?

**Comment by lxbsz** [comment]
> @lxbsz Please help me recall - the corrupted snaptrace issue was also supposed to be fixed in the clients too?

@vshankar Correct, but we only need to fix the kclient and the patch already get merged https://tracker.ceph.com/issues/61200.

For libcephfs it makes no sense because the libcephfs could always correctly decode the useless split_realms, but the kclient won't.

**Review by vshankar — APPROVED** [review]

**Comment by vshankar** [comment]
Running into lots of "failed to fetch package" issues with the run such as - https://pulpito.ceph.com/vshankar-2023-07-25_11:25:54-fs-wip-vshankar-testing-20230725.053049-testing-default-smithi/7350458/

The kernel build are available. @lxbsz suspects bug in teuthology. Will keep you updated.

**Review by vshankar — APPROVED** [review]
https://tracker.ceph.com/projects/cephfs/wiki/Main#28-JULY-2023

Nice work @lxbsz

**Pull request merged**